### ISSUE TITLE
status webpage - show which resource is exhausted?

### ISSUE THREAD

**Issue by keiranmraine** [VertebrateResequencing/wr#105]
It would be good if the status webpage could show which resource is "likely" exhausted (e.g. volumes, instances, storage, vCPU, RAM).

Not urgent, just nice to have.  Assuming this would be exposed via REST api automatically if added to the webpage.

**Comment by sb10** [comment]
Current develop already sends messages explaining exactly what quota ran out. These appear only on the web interface, and are a separate stream of messages to job status changes.

What is your use case for having them visible via the REST api? If you're building an interface for a human to look at, I'd suggest hooking in to the websocket just like wr's web interface does. This way you don't have to poll wr every x seconds to see if there's a new message.

I envisaged the REST api for programatic access to job creation and status when you weren't able to write or use a go client. Messages about quota make no difference to a program trying to run jobs; they're only informative to a human observer.

**Comment by keiranmraine** [comment]
REST wise I'm currently thinking about metrics for down the line.  I'd been thinking of not having my system add a new batch of jobs to the wr queue if it was blocked on resource.  For example we may have queued 100's of jobs within wr, but it wouldn't make sense to add any more if there's no resource left, primarily as should something happen to the wr manager I'd rather determine the state of a small number of jobs...

Other uses I can foresee:

* Ability to plot times where we are deficient on various resources depending on our use pattern
* Catch config/OpenStack issues - our system says it's doing nothing but WR says resource is exhausted

I don't see these as things I'd expect WR to be handling, but if the values are exposed they can be used for many things.

**Comment by sb10** [comment]
I think initially with lack of dependencies and now with lack of limiting to protect resources (the next major feature to be implemented) it made sense to reduce the number of jobs you add to wr, but I hope your mind set will change: wr is at its core a queue, and you're supposed to add everything you need done to that queue. The alternative that you have right now is that you hold some kind of queue yourself, but I suggest wr will be a more reliable queue (esp. if you're using S3 for its database backups).

As for messages via REST, I can see plotting them or otherwise keeping them as a useful use case that isn't supported by the web interface, so yes, I'll look in to making the messages queryable.

**Comment by keiranmraine** [comment]
I'm totally for dumping everything into WR longer term.  I'm just getting back into this side of things so if S3 db backup and multiple managers are ready when I'm at the point of pushing real data in I am less likely to hold data back.

**Comment by sb10** [comment]
Now implemented. The docs have been updated on the wiki: https://github.com/VertebrateResequencing/wr/wiki/REST-API
